SHS-5958: Bug: Inconsistent "Add Above" feature

codechefmarc opened the pull request:
# READY FOR REVIEW

## Summary
- Adds "Add between" buttons to paragraphs on Flexible Pages and Private Pages.

## Need Review By (Date)
2024-12-04

## Urgency
medium

## Steps to Test
1. Login to the site
2. Add a new Flexible Page or Private Page
3. Verify that there are "+ Add" buttons for paragraphs

![Screenshot 2024-12-04 at 11 42 51 AM](https://github.com/user-attachments/assets/3949c8cb-5c68-43c8-a5d1-f0d968b05883)

5. Add a paragraph
6. Verify that there is also an "+Add" button that gets added when hovering over a paragraph to add in between
7. Edit an existing Flexible Page or Private Page
8. Verify that the "+Add" buttons show up there as well

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208992447859410

Commit:
fix(SHS-5958): Fix issue with paragraphs features to allow add buttons on non gin/claro themes

Commit:
Merge branch '11.6.1-release' into SHS-5958--fix-add-above

Commit:
feat(SHS-5958): Add config and update hook to allow add paragraphs in between

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

Commit:
refactor(SHS-5958): Change how we are updating the add in between to config

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

Commit:
test(SHS-5958): Update test to refelect new buttons

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

Commit:
test(SHS-5958): Update testPostCard test for new add in between buttons

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

Commit:
test(SHS-5958): Update other tests for new add in between buttons

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

Commit:
test(SHS-5958): Add a wait in test to make sure JS add in between buttons load

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

Commit:
test(SHS-5958): Fix photo album test - photo album can come before or after text component, making tests work sometimes but not all the time

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

codechefmarc commented:
@cienvaras - This is ready for review. Note - I had to update some of the tests to click on the new add in-between buttons so those worked. Though I'm still getting some of those reoccurring test errors for unrelated things. I think I fixed the `testPhotoAlbum` one - it was because the photo album could be above or below the text area and sometimes it would fail it it was above it as the test was specifically calling item 1 (second paragraph). I changed it to use an XPath selector to look specifically for `stanford_gallery` but that test didn't run in the last test run so not sure. I'm out of hours for this week, so I can continue next week on these tests if needed. Thanks!

cienvaras commented on config/default/core.entity_form_display.node.hs_basic_page.default.yml:
```diff
@@ -79,22 +83,31 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
+  layout_builder__layout:
```
Not related to the current task, please revert.

codechefmarc replied:
Removed.

cienvaras commented on config/default/core.entity_form_display.node.hs_private_page.default.yml:
```diff
@@ -78,6 +83,12 @@ content:
           audio: 0
           video: 0
         default: link
+  layout_builder__layout:
```
Not related to the current task, please revert.

codechefmarc replied:
Removed.

cienvaras committed:
fix(shs-5958): fix video embed test

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras committed:
fix(shs-5958): fix private page tests

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras committed:
fix(shs-5958): fix video embed test

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras committed:
fix(shs-5958): fix private page tests

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras reviewed (changes requested):
@codechefmarc Thanks, looks great. There are just a couple of things to fix in the exported config.

I think I have an idea of why the tests are failing. I'll try a couple of fixes and let you know how it goes.

cienvaras committed:
fix(shs-5958): fixes in tests

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras committed:
fix(shs-5958): fix private page tests

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras committed:
fix(shs-5958): fix private page tests

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

Commit:
chore(SHS-5958): Remove unnecessary layout builder config

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras commented on config/default/core.entity_form_display.node.hs_basic_page.default.yml:
```diff
@@ -145,5 +152,4 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
-hidden:
-  layout_builder__layout: true
+hidden: {  }
```
@codechefmarc  Sorry, I didn't highlighted this in the review, but this needs to be reverted too, it's related to the other change you reverted before.

codechefmarc replied:
Reverted.

cienvaras commented on config/default/core.entity_form_display.node.hs_private_page.default.yml:
```diff
@@ -128,5 +133,4 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
-hidden:
-  layout_builder__layout: true
+hidden: {  }
```
@codechefmarc See previous comment.

codechefmarc replied:
Reverted.

cienvaras reviewed (changes requested):
@codechefmarc We missed a couple of changes that need to be reverted, please see inline comments. Thanks!

Commit:
Merge branch '11.6.1-release' into SHS-5958--fix-add-above

Commit:
Merge branch '11.6.1-release' into SHS-5958--fix-add-above

Commit:
chore(SHS-5958): Revert hidden layout builder layout

Commit:
chore(SHS-5958): Revert composer.lock to 11.6.1 version

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

cienvaras committed:
Merge branch '11.6.1-release' of github.com:SU-HSDO/suhumsci into SHS-5958--fix-add-above

cienvaras reviewed (approved):
LGTM!

@ahughes3 This is ready for you to review.

ahughes3 reviewed (approved):
Looks good!
